src/utils/encoders.ts

```diff
@@ -71,3 +122,37 @@ export const decodeOrder = (order: EncodedOrder): DecodedOrder => {
     ).toString(),
   };
 };
+
+/**
+ * Use the ratio between the z and the price geometric average to calculate what z
+ * value should be used for the other order in order to preserve correlation between
+ * the two orders - and provide the liquidity that should be used to achieve that.
+ */
+export const calculateRequiredLiquidity = (
+  knownOrder: DecodedOrder,
+  vagueOrder: DecodedOrder
+): string => {
+  const z: BigNumber = calculateCorrelatedZ(knownOrder);
+  const L: BigNumber = DecToBn(encodeRate(new Decimal(vagueOrder.lowestRate)));
+  const H: BigNumber = DecToBn(encodeRate(new Decimal(vagueOrder.highestRate)));
+  const M: BigNumber = DecToBn(
+    encodeRate(new Decimal(vagueOrder.marginalRate))
+  );
+
+  return z.mul(M.sub(L)).div(H.sub(L)).toString();
+};
+
+/**
+ * Use the ratio between the z and the price geometric average to calculate what z
+ * value should be used for the other order in order to preserve correlation between
+ * the two orders. This should be used when the other order has 0 liquidity - as z can't
+ * be determined otherwise.
```

Comment:
I'd change this:
> This should be used when the other order has 0 liquidity - as z can't be determined otherwise

To this:
> This function assumes that the other order has 0 liquidity

Because:
- If the other order has both liquidity and marge specified, then its `z` can be calculated internally
- If the other order has only marge specified, then its `z` can be calculated using this function
- If the other order has only liquidity specified, then its `z` can still be calculated, but in a different manner
- If the other order has neither liquidity nor marge specified, then its `z` can still be calculated, but in a different manner

Reply:
done